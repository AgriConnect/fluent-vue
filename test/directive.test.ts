import { createLocalVue, mount } from '@vue/test-utils'

import { FluentBundle, FluentResource } from '@fluent/bundle'
import ftl from '@fluent/dedent'

import FluentVue from '../src'

describe('directive', () => {
  let options: any
  let bundle: any

  beforeEach(() => {
    const localVue = createLocalVue()
    localVue.use(FluentVue)

    bundle = new FluentBundle('en-US', {
      useIsolating: false
    })

    const fluent = new FluentVue({
      bundle
    })

    options = {
      fluent,
      localVue
    }
  })

  it('translates text content', () => {
    // Arrange
    bundle.addResource(
      new FluentResource(ftl`
      link = Link text
      `)
    )

    const component = {
      data: () => ({
        name: 'John'
      }),
      template: `<a v-t="{ key: 'link' }" href="/foo">Fallback text</a>`
    }

    // Act
    const mounted = mount(component, options)

    // Assert
    expect(mounted).toMatchSnapshot()
  })

  it('can use parameters', () => {
    // Arrange
    bundle.addResource(
      new FluentResource(ftl`
      link = Hello {$name}
      `)
    )

    const component = {
      data: () => ({
        name: 'John'
      }),
      template: `<a v-t="{ key: 'link', arg: { name: 'John' } }" href="/foo">Fallback text</a>`
    }

    // Act
    const mounted = mount(component, options)

    // Assert
    expect(mounted).toMatchSnapshot()
  })
})
