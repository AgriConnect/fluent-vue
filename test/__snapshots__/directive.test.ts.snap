// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`directive can translate DOM attributes 1`] = `<a href="/foo" aria-label="Localized aria">Hello John</a>`;

exports[`directive can use parameters 1`] = `<a href="/foo">Hello John</a>`;

exports[`directive translates text content 1`] = `<a href="/foo">Link text</a>`;

exports[`directive warns about missing key arg 1`] = `<a href="/foo">Fallback text</a>`;

exports[`directive works without fallbacks 1`] = `<a aria-label="Localized aria">Hello John</a>`;
